cdap-docs/developer-manual/source/cloud-runtimes/profiles/assigning-profiles.rst

```diff
@@ -8,5 +8,75 @@
 Assigning Profiles
 ==================
 
-Describe how to assign a profile
+Profiles can be assigned to batch pipelines in a variety of ways. You can assign a default
+profile for all of CDAP as well as a default profile for a specific namespace. You can also
+assign a profile to a batch pipeline that will be used for runs that are manually started,
+and you can assign a profile to a pipeline schedule.
 
+The profile that is actually used at runtime is determined by walking the CDAP hierarchy
+from most general to most specific.
+
+instance > namespace > schedule/manual run
+
+If a profile is set on the schedule that triggered a run, that profile will be used.
+If no profile is set, the default profile for the namespace will be used. If no default
+is set on the namespace, the default for CDAP will be used. If no CDAP default is set,
+the native profile will be used.
+
+Similarly, if a pipeline is manually run and there is a profile assigned to that pipeline,
+that profile will be used.
+If no profile is set, the default profile for the namespace will be used. If no default
+is set on the namespace, the default for CDAP will be used. If no CDAP default is set,
+the native profile will be used.
+
+Note: If you have deployed your own custom applications and are using workflows, you
+can also use profiles to run your workflow in a cloud environment. Batch pipelines are
+just workflows, but have special UI support to make it easier to use profiles with them.
+ 
+Assigning Default Profiles
+--------------------------
+Default profiles can be assigned to a namespace and to the CDAP instance.
+This can be done through the UI by navigating to the Administration page and clicking
+on the star next to the profile.
+
+Alternatively, the :ref:`Preferences RESTful API <http-restful-api-preferences>` can be used
+to set default profiles. To set the default profile for all of CDAP, set a preference on the
+CDAP instance with key `profile` and value `system:<profile-name>`. To set the default profile for a
+namespace, set a preference on the desired namespace with key `profile` and value `<scope>:<profile-name>` 
```

Comment:
Is the key `profile`, or is it `system.profile.name`?

Reply:
sorry this is wrong, it should be system.profile.name